### PR TITLE
Fixes #3333. Application.UnGrabbedMouse event doesn't allow grab another view after exit.

### LINKED ISSUE #3333: Application.UnGrabbedMouse event doesn't allow grab another view after exit.
The issue is the `MouseGrabView` is always set to null at the end.

### PR DESCRIPTION
**Pull request by BDisp**
## Fixes

- Fixes #3333

## Proposed Changes/Todos

- [x] Sets MouseGrabView to null before call the event
- [x] Change unit test to proof
- [x] ScrollView now grab the mouse if any view want to grab it

## Pull Request checklist:

- [x] I've named my PR in the form of "Fixes #issue. Terse description."
- [x] My code follows the [style guidelines of Terminal.Gui](https://github.com/gui-cs/Terminal.Gui/blob/develop/.editorconfig) - if you use Visual Studio, hit `CTRL-K-D` to automatically reformat your files before committing.
- [x] My code follows the [Terminal.Gui library design guidelines](https://github.com/gui-cs/Terminal.Gui/blob/develop/CONTRIBUTING.md)
- [x] I ran `dotnet test` before commit
- [ ] I have made corresponding changes to the API documentation (using `///` style comments)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any poor grammar or misspellings
- [x] I conducted basic QA to assure all features are working

### CODE REVIEW

File: Terminal.Gui/Views/ScrollView.cs [hunk 1]
```diff
@@ -134,6 +136,26 @@ public ScrollView ()
...
+        if (parent is { })
+        {
+            var supView = parent.SuperView;
+
+            while (supView is { })
+            {
+                if (supView == _contentView)
+                {
+                    Application.GrabMouse (this);
+                }
```
Comment (tig) [inline Terminal.Gui/Views/ScrollView.cs:152]: Are you sure there shouldn't be a `break;` here to exit the loop?

**Review by tig — CHANGES_REQUESTED** [review]
It seems to me while loop should be cut short once a matching view is found?

File: Terminal.Gui/Views/ScrollView.cs (hunk 1 above)
Comment (BDisp) [reply Terminal.Gui/Views/ScrollView.cs:152]: You are absolutely right. When I did that I think to my self, don't forget to break the loop if found it. Damn and I forgot. Thanks.

**Review by tig — APPROVED** [review]
Good stuff! Well done.

**Pull request merged**